process/block/sovereign/outgoingOperations_test.go

```diff
@@ -25,115 +25,376 @@ func createEvents() []SubscribedEvent {
+				return &sovereign.EventData{}, nil
+			},
+			DeserializeTokenDataCalled: func(_ []byte) (*sovereign.EsdtTokenData, error) {
+				return &sovereign.EsdtTokenData{}, nil
+			},
+			SerializeOperationCalled: func(operation sovereign.Operation) ([]byte, error) {
+				return nil, fmt.Errorf("serialize operation error")
+			},
+		}
+
+		outgoingTxData, err := outgoingOps.CreateOutgoingTxsData(logs)
+		require.Nil(t, outgoingTxData)
+		require.Equal(t, "serialize operation error", err.Error())
+	})
+}
+
+func TestOutgoingOperations_CheckValidity(t *testing.T) {
+	t.Parallel()
+
+	dataCodec := &mock.DataCodecMock{}
+	topicsChecker := &mock.TopicsCheckerMock{
+		CheckValidityCalled: func(topics [][]byte) error {
+			return fmt.Errorf("invalid")
+		},
+	}
 
 	events := []SubscribedEvent{
 		{
-			Identifier: identifier1,
+			Identifier: []byte("deposit"),
 			Addresses: map[string]string{
-				string(addr1): string(addr1),
-				string(addr2): string(addr2),
+				"addr1": "addr1",
 			},
 		},
+	}
+
+	args := ArgsOutgoingOperations{
+		SubscribedEvents: events,
+		DataCodec:        dataCodec,
+		TopicsChecker:    topicsChecker,
+	}
+	creator, _ := NewOutgoingOperationsFormatter(args)
+
+	logs := []*data.LogData{
 		{
-			Identifier: identifier2,
-			Addresses: map[string]string{
-				string(addr3): string(addr3),
+			LogHandler: &transactionData.Log{
+				Address: nil,
+				Events: []*transactionData.Event{
+					{
+						Address:    []byte("addr1"),
+						Identifier: []byte("deposit"),
+						Topics:     [][]byte{[]byte("topic1")},
+						Data:       []byte("data"),
+					},
+				},
 			},
+			TxHash: "",
 		},
 	}
 
-	roundHandler := &testscommon.RoundHandlerMock{
-		IndexCalled: func() int64 {
-			return 123
+	outgoingTxData, err := creator.CreateOutgoingTxsData(logs)
+	require.Nil(t, outgoingTxData)
+	require.Error(t, err, "invalid")
+}
+
+func TestOutgoingOperations_CreateOutgoingTxData(t *testing.T) {
+	t.Parallel()
+
+	addr1 := []byte("addr1")
+	addr2 := []byte("addr2")
+	addr3 := []byte("addr3")
+
+	identifier1 := []byte("deposit")
+	identifier2 := []byte("send")
+
+	tokenData1 := []byte("tokenData1")
+	topic1 := [][]byte{
+		[]byte("deposit"),
+		[]byte("rcv1"),
+		[]byte("token1"),
+		[]byte("nonce1"),
+		tokenData1,
+	}
+	data1 := []byte("data1")
+
+	evData := &sovereign.EventData{
+		Nonce: 1,
+		TransferData: &sovereign.TransferData{
+			GasLimit: 20000000,
+			Function: []byte("add"),
+			Args:     [][]byte{big.NewInt(20000000).Bytes()},
 		},
 	}
 
+	addr0, _ := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+	amount := new(big.Int)
+	amount.SetString("123000000000000000000", 10)
+	tokenData := sovereign.EsdtTokenData{
+		TokenType:  0,
+		Amount:     amount,
+		Frozen:     false,
+		Hash:       make([]byte, 0),
+		Name:       make([]byte, 0),
+		Attributes: make([]byte, 0),
+		Creator:    addr0,
+		Royalties:  big.NewInt(0),
+		Uris:       make([][]byte, 0),
+	}
```

Comment:
You don't need to set addr0, amount, etc.
Just create a simple tokenData with 2-3 relevant fields